### PR TITLE
fix: ShowAlert with same texts, when invoked from different triggers are combined 

### LINKED ISSUE #16135: [Bug]: ShowAlert with same texts, when invoked from different triggers are combined
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://user-images.githubusercontent.com/91450662/185391846-601626d0-1a94-4c9c-b9ad-7c5fc00eafd2.mov

### Steps To Reproduce

1. D&D a button & update the onclick event to `{{JSObject1.myFun1();
showAlert('Clicked!','success')}}`
2. Define a JS function with `myFun1: () => {
		showAlert('Clicked!', "success");
	}`
3. Now trigger the onClick on the button
4. **Observe: as the texts are same in both showAlerts - there is only one toast which could be confusing user**
5. **Expected: As many number of toasts as defined by user should appear**

### Public Sample App

_No response_

### Version

Prod/Release v1.7.11

### PR DESCRIPTION
**Pull request by srix**
## Description
The toaster component, part of the react design system, generates a unique id for each toast. The id is generated based on a content string and the options Json. When the content and options are identical, duplicate ids are generated. Toaster ignores any toast with duplicate IDs.  Explicitly passing a unique id overrides the internal id generation logic. This ensures that all toasters are displayed, even if the content and options are identical.

#### PR fixes following issue(s)
Fixes #16135 

#### Media
None

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x ] Manual
- [ x] Cypress
>
>
#### Test Plan
cypress test case is defined in 
`app/client/cypress/e2e/Regression/ServerSide/JsFunctionExecution/PlatformFn_spec.ts/2.Bug 16135 ShowAlert with same texts, when invoked from different triggers are combined
`

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by arunvjn — APPROVED** [review]
Looks good!

**Comment by shwetha-ramesh** [comment]
@srix QA Done. Working as expected. Moving to Done
image

File: app/client/cypress/e2e/Regression/ServerSide/JsFunctionExecution/PlatformFn_spec.ts [hunk 1]
```diff
@@ -144,4 +145,29 @@ describe("Tests functionality of platform function", () => {
...
+      {
+        paste: true,
+        completeReplace: true,
+        toRun: false,
+        shouldCreateNewJSObj: true,
+        prettify: false,
+      },
+    );
+    agHelper.Sleep();
+    jsEditor.SelectFunctionDropdown("showTwoSameToastMessageAlerts");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/JsFunctionExecution/PlatformFn_spec.ts:167]: Above JS object creation already has the function selected. So this step is not needed.

Comment (srix) [reply app/client/cypress/e2e/Regression/ServerSide/JsFunctionExecution/PlatformFn_spec.ts:167]: Though it is better to be explicit. I will accept the comment. and make changes

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ServerSide/JsFunctionExecution/PlatformFn_spec.ts]: No, when only one function is present in Js Object - it is expected to be updated in dropdown. 
We are also verifying that is true by simply running & not needing to select it. 
I dont see a reason for it to be explicit except for readability of test code, hence the ask.

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes.

**Comment by srix** [comment]
Label Checker / Check for Test Plan Approved label (pull_request)  Failed
I think QA team needs to add the label . @Aishwarya-U-R can you point me to the right person

**Comment by Aishwarya-U-R** [comment]
Since cypress test TableV2TextPagination_spec is known failure, merging this PR in.

**Pull request merged**